pkg/tests/tasks/security/authorization/http_test.go

```diff
@@ -16,105 +16,216 @@ package authorization
-			t.Errorf("GET policy failed. Got response: %s", string(body))
-			log.Log.Errorf("GET policy failed. Got response: %s", string(body))
-		} else {
-			log.Log.Infof("Got expected page.")
-		}
-		util.CloseResponseBody(resp)
+// TestAuthorizationHTTPTraffic validates authorization policies for HTTP traffic.
+func TestAuthorizationHTTPTraffic(t *testing.T) {
+	test.NewTest(t).Id("T20").Groups(test.Full, test.ARM, test.InterOp).Run(func(t test.TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+
+		ns := "bookinfo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+			oc.MergePatch(t, meshNamespace,
+				fmt.Sprintf(`smcp/%s`, smcpName),
+				"merge",
+				`{"spec":{"security":{"dataPlane":{"mtls":false},"controlPlane":{"mtls":false}}}}`,
+			)
+			oc.WaitSMCPReady(t, meshNamespace, smcpName)
+		})
+
+		t.Log("This test validates authorization policies for HTTP traffic.")
+		t.Log("Doc reference: https://istio.io/v1.14/docs/tasks/security/authorization/authz-http/")
+
+		t.LogStep("Enable Service Mesh Control Plane mTLS")
+		oc.MergePatch(t, meshNamespace,
+			fmt.Sprintf(`smcp/%s`, smcpName),
+			"merge",
+			`{"spec":{"security":{"dataPlane":{"mtls":true},"controlPlane":{"mtls":true}}}}`,
+		)
+		oc.WaitSMCPReady(t, meshNamespace, smcpName)
+
+		t.LogStep("Install bookinfo with mTLS")
+		app.InstallAndWaitReady(t, app.BookinfoWithMTLS(ns))
+		ProductPage_URL := app.BookinfoProductPageURL(t, meshNamespace)
+
+		t.NewSubTest("deny all http traffic to bookinfo").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, DenyAllPolicy)
+			})
+			t.LogStep("Apply policy that denies all HTTP requests to bookinfo workloads")
+			oc.ApplyString(t, ns, DenyAllPolicy)
+
+			t.LogStep("Verify that GET request is denied")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				Curl.Request(t,
+					ProductPage_URL,
+					nil,
+					assert.ResponseContains("RBAC: access denied"),
+				)
+			})
+		})
+
+		t.NewSubTest("only allow HTTP GET request to the productpage workload").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, ProductpageGETPolicy)
+				oc.DeleteFromString(t, ns, DenyAllPolicy)
+			})
+			t.LogStep("Apply policies that allow access with GET method to the productpage workload and deny requests to other workloads")
+			oc.ApplyString(t, ns, DenyAllPolicy)
+			oc.ApplyString(t, ns, ProductpageGETPolicy)
+
+			t.LogStep("Verify that GET request to the productpage is allowed and fetching other services is denied")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				Curl.Request(t,
+					ProductPage_URL,
+					nil,
+					assert.ResponseContains("Error fetching product details"),
+				)
+				Curl.Request(t,
+					ProductPage_URL,
+					nil,
+					assert.ResponseContains("Error fetching product reviews"),
+				)
```

Comment:
This can be a single request with two asserts:
```suggestion
				Curl.Request(t,
					ProductPage_URL,
					nil,
					assert.ResponseContains("Error fetching product details"),
					assert.ResponseContains("Error fetching product reviews"),
				)
```